### PR TITLE
Refactor MeiliSearch Client

### PR DESCRIPTION
**Pull request by bidoubiwa**
fixes: #146 
fixes: #147

## Config removed from MeiliSearch client constructor

To avoid variable conflicts between multiple libraries and also to conform to practices we observed in other repos, config is not used anymore as a parameter to create a MeiliSearch client.

### Before: 

```swift
MeiliSearch(Config(hostURL: String?, apiKey: String?, session: session?)
```

Example:
```swift
client = try! MeiliSearch(Config(hostURL: nil, apiKey: nil, session: session))
```

### After

```swift
MeiliSearch(_ hostURL: String,_ apiKey: String? = nil,_ session: URLSessionProtocol? = nil)
```

Example:
```swift
client = try! MeiliSearch("http://localhost:7700")
```
```swift
client = try! MeiliSearch("http://localhost:7700", "masterKey")
```
```swift
client = try! MeiliSearch("http://localhost:7700", "masterKey", session)
```

### Parameter naming

For normalization between SDK's parameters are not named on function call. 

```swift
MeiliSearch(hostURL: "http://localhost:7700") // Not working
```

## Mandatory and optional parameters

- HostURL is now mandatory as a parameter.
- ApiKey is __optional__ and has a default value of `nil`
- Session is __optional__ and has a default value of `nil`

## Default host removed

Previously the client could be initialised with no information. This was because we provided a default host in the Config struct: `http://localhost:7700`. 

The following was previously possible and is not anymore:

```swift
// We do not use config anymore. Default does not exist and HostURL is now mandatory
client = try! MeiliSearch(Config.default(apiKey: "masterKey"))
```

```swift
// HostURL is mandatory 
let client = try! MeiliSearch()
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,8 +78,8 @@ import MeiliSearch
 
 func searchForMovies() {
 
-    // Create a new client instance of MeiliSearch with the default host.
-    let client = try! MeiliSearch()
+    // Create a new client instance of MeiliSearch a running host.
```
Comment (bidoubiwa) [inline README.md:81]:
```suggestion
    // Create a new client instance of MeiliSearch.
```

File: Tests/MeiliSearchIntegrationTests/DocumentsTests.swift [hunk 2]
```diff
@@ -172,6 +172,7 @@ class DocumentsTests: XCTestCase {
             case .success:
                 XCTFail("Document has been found while it should not have")
             case .failure:
+                print(result)
```
Comment (bidoubiwa) [inline Tests/MeiliSearchIntegrationTests/DocumentsTests.swift:175]:
```suggestion
```

File: Tests/MeiliSearchIntegrationTests/DocumentsTests.swift [hunk 3]
```diff
@@ -372,6 +373,7 @@ class DocumentsTests: XCTestCase {
             case .success:
                 XCTFail("Movie should not exist")
             case .failure:
+                print(result)
```
Comment (bidoubiwa) [inline Tests/MeiliSearchIntegrationTests/DocumentsTests.swift:376]:
```suggestion
```

**Review by alallema — APPROVED** [review]
LGTM! 🎉

**Review by alallema — APPROVED** [review]

File: .code-samples.meilisearch.yaml [hunk 4]
```diff
@@ -847,7 +847,7 @@ getting_started_search_md: |-
   ```
   Example:
   ```swift
-    let client = try! MeiliSearch(Config(hostURL: "http://localhost:7700"))
+    let client = try! MeiliSearch(Config("http://localhost:7700")
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:850]: A `)` is missing

File: README.md [hunk 5]
```diff
@@ -79,7 +79,7 @@ import MeiliSearch
 func searchForMovies() {
 
     // Create a new client instance of MeiliSearch a running host.
-    let client = try! MeiliSearch(hostURL: "http://localhost:7700")
+    let client = try! MeiliSearch("http://localost:7700")
```
Comment (curquiza) [inline README.md:82]:
```suggestion
    let client = try! MeiliSearch("http://localhost:7700")
```

**Review by curquiza — CHANGES_REQUESTED** [review]

**Comment by curquiza** [comment]
Well done!!!

File: .code-samples.meilisearch.yaml (hunk 4 above)
Comment (bidoubiwa) [reply .code-samples.meilisearch.yaml:850]:
```suggestion
    let client = try! MeiliSearch("http://localhost:7700")
```

Thanks, config should have been removed as well :)

**Review by curquiza — APPROVED** [review]

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**